### ISSUE TITLE
Accumulating many .foz files with Gamescope

### ISSUE THREAD

**Issue by cchulo** [ValveSoftware/Fossilize#212]
So I am running Emulationstation-DE (non-steam game), Resident Evil HD, and Nier Automata all with Gamescope. I have been noticing that every time I start up with gamescope it creates a new .foz file inside of the steamapps/shadercache/fozpipelinesv6 directory. It looks like the way it is creating them is a bug because the hash is the same, but it keeps making copies. Take RE HD for example:

it is executed with `gamescope -H 1440 -h 1440 -r 60 -f -- %command%`
The Foz files that are created every time look like:
![image](https://user-images.githubusercontent.com/9343504/212772998-6795a1f9-79ab-45e7-b8d7-391ac7465248.png)

With emulationstation-DE, something that's been installed longer than RE HD and has been used quite a bit:
![image](https://user-images.githubusercontent.com/9343504/212773734-08db47e1-37d2-494a-9941-87e1a934939b.png)

Note that for ES-DE the hash changed at some point, this may have been due to me installing new GPU drivers

With Nier its the same situation, a new .foz file everytime I launch the game

All the .foz files are the same size, is this a bug? This only happens when games are executed with gamescope. It does not matter if its a non-steam game or not

**Comment by cchulo** [comment]
I don't think this is an issue with Fossilize, it may be with steam runtime, though I will keep this issue open if anyone has insight as to what may be causing this

**Comment by kisak-valve** [comment]
Hello @cchulo, if you find a decent hint that this issue is rooted in another part of the larger set of Steam-related projects, please add the note here and give me a ping requesting for this issue report to be transferred to the other issue tracker so that we keep the discussion of this issue centralized and skip making duplicate issue reports across multiple issue trackers.

**Comment by cchulo** [comment]
@kisak-valve thank you, I was considering opening an issue with steam-runtime project, but at this point I am a bit stumped, I am not sure what is responsible for creating these files. I looked everywhere in github for the keyword "steamapprun", might be a closed source project that is responsible.

So today I discovered that Quake, a game I have had no previous issues with running under KWin, and therefore there is no need to use gamescope with, has also been accumulating .foz files. In fact, I don't even use anything with it other than proton 7.0.5

This leads me to believe that gamescope itself is not responsible, but perhaps gamescope might be making steam think the shader cache is invalid everytime a game is launched (?)

The other games I mentioned that are all exhibiting these symptoms (emulationstation, Nier, and RE HD) only produce these spare foz files whenever they are executed with gamescope, but are just fine when I run them without it.

So in summary, the gamescope executable and Quake seem to be causing an issue.

Might be relevant to mention that I am using nvidia drivers (historically has not played nice with linux at times), here are my specs:

OS: EndeavourOS
LinuxKERNEL: 6.1.7-arch1-1
CPU: AMD Ryzen 5 5600X 6-Core
GPU: NVIDIA GeForce RTX 3080
GPU DRIVER: NVIDIA 525.85.05
RAM: 32 GB

What do you recommend?

**Comment by danginsburg** [comment]
This is definitely a Steam issue and not a Fossilize issue specifically.  There is a cache merging phase that Steam runs on application exit that apparently isn't happening when launched directly from gamescope.  It would have merged these individual .foz files into a single .foz file (de-duplicating) and deleting the individual ones.

**Comment by cchulo** [comment]
I see, thanks @danginsburg is the steam-runtime repo appropriate place for this issue?

**Comment by Trayshar** [comment]
I observe the same behaviour in Deep Rock Galactic. Multiple small .foz files, and they are never merged into a single one. I do not use gamescope, however, should this get it's own issue? And what's the status of this issue?
